product/views.py

```diff
@@ -1,3 +1,58 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.views import View
+from django.http import JsonResponse
+from django.db.models import Prefetch
+from user.models import Creator,User
+from request.models import CourseRequest
+from product.models import (
+    Course,
+    CourseStatus,
+    Category,
+    SubCategory,
+    ProductImage,
+    Product,
+    ProductLike,
+)
+
+#클래스 좋아요 기능
+class CourseLikeView(View):
+    #TODO : ADD  @id_auth
+    def post(self, request, product_pk):
+        print(product_pk)
+        try:
+           # user = request.user
+            user   = User.objects.get(id = 1)
+            course = Product.objects.get(id = product_pk)
+
+            like, course_like = ProductLike.objects.get_or_create(user = user, product = course)
+            if not course_like:
+                return JsonResponse({'message':'ALREADY_LIKE'},status = 400)
+            return JsonResponse({'message':'SUCCESS'}, status = 201)
```

Comment:
중복 체크를 해야되는 경우 exists() method를 사용하는게 실행 속도 측면에서 빠릅니다.
```suggestion
            if ProductLike.objects.filter(user = user, product = course).exists()
                    return JsonResponse({'message':'ALREADY_LIKE'},status = 400)
            ProductLike.objects.create(..)
            return JsonResponse({'message':'SUCCESS'}, status = 201)
```

Reply:
반영했습니다 감사합니다